src/devices/Dhtxx/DhtSensor.cs

```diff
@@ -0,0 +1,218 @@
+        
+        /// <summary>
+        /// Get the last read temperature
+        /// </summary>
+        /// <remarks>
+        /// If last read was not successfull, it returns double.MaxValue
+        /// </remarks>
+        public double Temperature
+        {
+            get {
+                if (dhtType == DhtType.Dht11)
+                    return GetTempDht11();
+                else
+                    return GetTempDht22();
+            }
+        }
+        
+        /// <summary>
+        /// Get the last read humidity
+        /// </summary>
+        /// <remarks>
+        /// If last read was not successfull, it returns double.MaxValue
+        /// </remarks>
+        public double Humidity
+        {
+            get {
+                if (dhtType == DhtType.Dht11)
+                    return GetHumidityDht11();
+                else
+                    return GetHumidityDht22();
+            }
+        }
+
+        /// <summary>
+        /// Create a DHT sensor
+        /// </summary>
+        /// <param name="pin">The pin number (GPIO number)</param>
+        /// <param name="dhtType">The DHT Type, eitherDht11 or Dht22</param>
+        public DHTSensor(int pin, DhtType dhtType)
+        {
+            this.pin = pin;
+            this.dhtType = dhtType;
+        }
+
+        /// <summary>
+        /// Start a reading
+        /// </summary>
+        /// <returns>
+        /// <c>true</c> if read is successfull, otherwise <c>false</c>.
+        /// </returns>
+        public bool ReadData()
+        {
+            using (GpioController controller = new GpioController())
+            {                
+                // Set the max value for waiting micro second
+                // 27 = debug
+                // 99 = release
+                byte waitMS = 99;
+                #if DEBUG
+                    waitMS = 27;
+                 #endif
+                controller.OpenPin(pin);
+
+                DelayMS delayMS= new DelayMS();
+                PinValue lststate = PinValue.High; 
+                uint counter = 0; 
+                byte j = 0, i; 
+                for (i = 0; i < 5; i++) 
+                    dht11_val[i] = 0; 
+
+                // write on the pin
+                controller.SetPinMode(pin, PinMode.Output);
+                controller.Write(pin, PinValue.Low); 
+                //wait 18 milliseconds
+                delayMS.Wait(18); 
+                controller.Write(pin, PinValue.High); 
+                delayMS.Wait(0.02); 
+                controller.SetPinMode(pin, PinMode.Input);
+                for (i = 0; i < MAX_TIME; i++) 
+                { 
+                    counter = 0; 
+                    while (controller.Read(pin) == lststate){ 
+                        counter++; 
+                        // This wait about 1 micro second
+                        // No other way to do it for such a precision
+                        for(byte wt=0; wt< waitMS; wt++)
```

Comment:
I'm wondering if this could be done as a separate function which is inline and has some optimizations turned off/on explicitly. I think it probably is ok for now but maybe worth filing issue to provide something like that and perhaps work with the runtime team to make sure we can do it correctly